Run sfc_climo_gen utility on Hera

GeorgeGayno-NOAA opened the pull request:
## DESCRIPTION OF CHANGES: 
Add script to run the sfc_climo_gen utility on Hera. Add common logic to a new
script.

## TESTS CONDUCTED: 
Tested on WCOSS2 and Hera

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #788

GeorgeGayno-NOAA committed:
Baseline initial version of Hera script.

Fixes #788.

GeorgeGayno-NOAA committed:
Update the wcoss2 script for 'frac oro' data.

Fixes #788.

GeorgeGayno-NOAA committed:
Move common logic into a new script - sfc_gen.sh

Fixes #788.

GeorgeGayno-NOAA committed:
Update Hera script to invoke sfc_gen.sh.

Fixes #788.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.

JeffBeck-NOAA commented on util/sfc_climo_gen/run.hera.sh:
```diff
@@ -0,0 +1,29 @@
+# Set the configuration variables in sfc_gen.sh. Then
+# run this script as follows: 'sbatch $script'
+#------------------------------------------------------------
+
```
The pre-existing 'grid' and 'orography' files need to be staged in FIX_FV3=${BASE_DIR}/fix/orog/C${res}, right?  Can that info be added to the comment block above, including the names of the files?

GeorgeGayno-NOAA replied:
Added more comments at 5baa903.

JeffBeck-NOAA commented on util/sfc_climo_gen/run.hera.sh:
```diff
@@ -0,0 +1,29 @@
+#------------------------------------------------------------
+# Run the sfc_climo_gen program stand-alone on Hera using
+# pre-exiting 'grid' and 'orography files.
+#
```
'orography', missing second quotation.

GeorgeGayno-NOAA replied:
Fixed.

JeffBeck-NOAA commented on util/sfc_climo_gen/run.wcoss2.sh:
```diff
@@ -22,88 +23,6 @@ export APRUN_SFC="mpiexec -n 24 -ppn 24 -cpu-bind core"
-rm -fr $WORK_DIR $SAVE_DIR
-
-${BASE_DIR}/ush/sfc_climo_gen.sh
+$PBS_O_WORKDIR/sfc_gen.sh
```
Why was the block of code below removed?  I assume we want to keep this functionality on WCOSS2, correct?

GeorgeGayno-NOAA replied:
The common logic - including the invocation of `sfc_climo_gen.sh` - was moved to new script `sfc_gen.sh`.  That new script is called from the hera and wcoss2 (and any future) driver scripts.

JeffBeck-NOAA replied:
Sorry, I got confused when comparing the wrong files.  Looks good!

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.

GeorgeGayno-NOAA committed:
Fix typo.

Fixes #788.

GeorgeGayno-NOAA committed:
Update comments.

Fixes #788.

JeffBeck-NOAA commented on util/sfc_climo_gen/run.wcoss2.sh:
```diff
@@ -2,9 +2,11 @@
-# pre-exiting 'grid' and 'orography files.
+# pre-exiting 'grid' and 'orography files. See the
+# sfc_gen.sh script for details.
 #
```
One more quotation here.

GeorgeGayno-NOAA replied:
Fixed. Thanks.

GeorgeGayno-NOAA committed:
Update logic for regional grids. Create a local 'fixed'
directory to store links to the input files using names
that are recognized by the program. Previously, the
code tried to create links in the official 'fixed' directory
but most users won't have permissions to do that.

Fixes #788.

GeorgeGayno-NOAA committed:
Fix typo.

Fixes #788.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.

GeorgeGayno-NOAA committed:
Add new namelist varaible - vegsoil_frac - to sfc_gen.sh

Fixes #788.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/hera_scg

Fixes #788.
